lib/hsa/activity_prof.h

```diff
@@ -113,27 +113,45 @@ class ActivityProf {
         }
     }
 
-    // Activity callback routine
-    void callback(const command_id_t& command_id, const uint64_t& begin_ts, const uint64_t& end_ts, const size_t& bytes = 0) {
+    template <class T>
+    inline void callback_gpu(T* obj, const size_t& bytes = 0) {
+        if (_enabled == true) {
+            const command_id_t command_id = obj->getCommandKind();
+            uint64_t start = obj->getBeginTimestamp();
+            uint64_t end   = obj->getEndTimestamp();
+            callback(command_id, start, end, bytes);
+        }
+    }
+
+    template <class T>
+    inline void callback_sys(T* obj, const size_t& bytes = 0) {
```

Comment:
These two templates should have a more meaningful name and they aren't the actual callback functions.  How about renaming them to `report_gpu_timestamps` and `report_system_ticks` respectively?